design/safe-disruption-of-nodes-with-osds.md

```diff
@@ -0,0 +1,30 @@
+# Safe disruption of nodes with osds
+
+
+External operators and agents that provision,fence, or upgrade nodes should be able to use the kubernetes api to know how not to disrupt storage in the course of their operation.
+
+External agents should only bring down one failure domain at once.
+
+
+## Assumptions:
+- The failure domain is node/host(or higher)
+- To preserve availability, it is enough to ensure that only one host-that-has osds is down at once. 
+- It is fine for the operator to constantly ping the cluster for it’s status.
+
+## Proposed changes to `rook-ceph`:
+
+- The nodes with osds can be labeled with `ceph.rook.io/has-osds=true`
+- The nodes with osds can be annotated with the list of clusters that have osds on it.
+    - The key can be `ceph.rook.io/tenant-clusters`
```

Comment:
Same answer as previous comment
> I'm proposing that rook expose the information of which clusters have osds on which nodes so that an external infra agent performing node-level actions not directly related to rook can do it while maintaining storage availability.

Reply:
So in this example, `tenant-cluster` is the name of the Ceph cluster right?